sim/rogue/combat/revealing_strike.go

```diff
@@ -24,18 +25,23 @@ func (comRogue *CombatRogue) registerRevealingStrike() {
 			ActionID: actionID,
 			Duration: 15 * time.Second,
 
-			// Technically this _could_ cause problems in a target swapping situation, but it's good enough.
 			OnGain: func(aura *core.Aura, sim *core.Simulation) {
-				comRogue.Eviscerate.DamageMultiplier *= multiplier
-				comRogue.Envenom.DamageMultiplier *= multiplier
-				comRogue.Rupture.DamageMultiplier *= multiplier
+				if !isApplied {
+					comRogue.Eviscerate.DamageMultiplier *= multiplier
+					comRogue.Envenom.DamageMultiplier *= multiplier
+					comRogue.Rupture.DamageMultiplier *= multiplier
+					isApplied = true
+				}
+			},
+			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+				comRogue.Eviscerate.DamageMultiplier /= multiplier
+				comRogue.Envenom.DamageMultiplier /= multiplier
+				comRogue.Rupture.DamageMultiplier /= multiplier
+				isApplied = false
+				aura.Deactivate(sim)
```

Comment:
Revealing Strike is applied to each target as a debuff right? This implementation flat buffs the rogue's next strike regardless of target. I think the right way to do this would be to register a debuff on the target, then in the various finisher calculations, check if the buff is active and apply the multiplier